src/plugins/widgets/search/Search.tsx

```diff
@@ -72,6 +179,60 @@ class Search extends React.PureComponent<Props & InjectedIntlProps, State> {
 
     return searchEngine.search_url.replace('{searchTerms}', query);
   }
+
+  /**
+   * Retrieves suggestiondata from google
+   */
+  private getSuggestionData() {
+    const { query, getSuggestionData } = this.state;
+
+    if (!getSuggestionData || !this.props.active) {
+      return;
+    }
+
+    // To get the most recent result
+    const executionTime = window.performance.now();
+
+    if (!query) {
+      this.setSuggestions(executionTime, undefined);
+      return;
+    }
+
+    getSuggestions(query, suggestions => {
```

Comment:
Injecting the `suggest_url` from the engine config here to allowing looking up for different search engine providers.

You could extract this engine lookup to a lifecycle method and store it in the component to save duplicated code:
```
     const searchEngine = engines
       .find(engine => engine.key === this.props.engine)
       || engines[0];
```

Reply:
I did move the url into `engines.json` but I don't think the function `getSuggestions` will be reusable for suggestions form other providers because it's designed to get suggestions from google.